packages/preview/src/browser/preview-contribution.ts

```diff
@@ -0,0 +1,194 @@
+import { EDITOR_CONTEXT_MENU, EditorManager, TextEditor } from '@theia/editor/lib/browser';
+import { CommandContribution, CommandRegistry, Command, MenuContribution, MenuModelRegistry, CommandHandler, Disposable } from "@theia/core/lib/common";
+import { DisposableCollection } from '@theia/core';
+import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import URI from '@theia/core/lib/common/uri';
+import { Position } from 'vscode-languageserver-types';
+import { PreviewWidget, PREVIEW_WIDGET_FACTORY_ID } from './preview-widget';
+import { PreviewWidgetFactory } from './preview-widget-factory';
+import { PreviewHandlerProvider } from './preview-handler';
+
+export namespace PreviewCommands {
+    export const OPEN: Command = {
+        id: 'preview:open',
+        label: 'Open Preview'
+    };
+}
+
+@injectable()
+export class PreviewContribution implements CommandContribution, MenuContribution, OpenHandler, FrontendApplicationContribution {
+
+    readonly id = 'preview';
+    readonly label = 'Preview';
+
+    protected readonly editorDisposables = new DisposableCollection();
+    protected readonly previewDisposables = new DisposableCollection();
+
+    @inject(FrontendApplication)
+    protected readonly app: FrontendApplication;
+
+    @inject(WidgetManager)
+    protected readonly widgetManager: WidgetManager;
+
+    @inject(EditorManager)
+    protected readonly editorManager: EditorManager;
+
+    @inject(PreviewHandlerProvider)
+    protected readonly previewHandlerProvider: PreviewHandlerProvider;
+
+    @inject(PreviewWidgetFactory)
+    protected readonly previewWidgetFactory: PreviewWidgetFactory;
+
+    onStart() {
+        this.previewWidgetFactory.onWidgetCreated(newPreviewWidget => {
+            this.previewDisposables.dispose();
+            this.previewDisposables.push(this.registerOpenOnDoubleClick(newPreviewWidget));
+        });
+        this.syncWithCurrentEditor();
+    }
+
+    protected async syncWithCurrentEditor(): Promise<void> {
+        this.editorManager.onCurrentEditorChanged(async editorWidget => {
+            if (!editorWidget) {
+                return;
+            }
+            this.editorDisposables.dispose();
+            const editor = editorWidget.editor;
+            this.editorDisposables.push(editor.onCursorPositionChanged(position =>
+                this.synchronizeSelectionToPreview(editor, position))
+            );
+            const previewWidget = this.getPreviewWidget();
+            if (previewWidget) {
+                this.editorDisposables.push(this.synchronizeSelectionToEditor(previewWidget, editor));
+            }
+        });
+    }
+
+    protected getPreviewWidget(): PreviewWidget | undefined {
+        // note: this also ensures, we get resotored widgets
+        const previewWidget = this.widgetManager.getWidgets(PREVIEW_WIDGET_FACTORY_ID).slice(-1)[0] as PreviewWidget;
```

Comment:
though, this will be obsolete with the "multi widget" scenario, the original intention to use `WidgetManager.getWidgets` was to register event listeners for the instance, that was restored after start. I need to admin, that I switched to a widget factory which can now notify about created widgets, but I left this `getPreviewWidget`.

also, I did not yet apply Anton's hint to test, if widget is attached, in order to get finally rid of `WidgetManager.getWidgets` calls.